### PR TITLE
GHA - Replace Vercel bash deploy script by "deploy-on-vercel" github action

### PR DESCRIPTION
**Pull request by Demmonius**
Reduce the complexity of the Vercel deployment script (GitHub Actions) by using a dedicated action instead of the existing bash script.

Makes the code much more readable, remove code duplication (x3) in various actions.

### CODE REVIEW

File: .github/workflows/deploy-vercel-staging.yml [hunk 1]
```diff
@@ -254,8 +177,8 @@ jobs:
         with:
           variables: |
             CUSTOMER_REF_TO_DEPLOY=${{ env.CUSTOMER_REF_TO_DEPLOY }}
-            VERCEL_DEPLOYMENT_URL=${{ env.VERCEL_DEPLOYMENT_URL }}
             VERCEL_DEPLOYMENT_DOMAIN=${{ env.VERCEL_DEPLOYMENT_DOMAIN }}
+            VERCEL_DEPLOYMENT_URL=${{ format('https://{0}', env.VERCEL_DEPLOYMENT_DOMAIN) }}
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: Shouldn't the VERCEL_DEPLOYMENT_URL and VERCEL_DEPLOYMENT_DOMAIN both be returned by the Action?

Comment (Vadorequest) [reply .github/workflows/deploy-vercel-staging.yml]: That'd avoid doing the above which isn't dev-friendly

File: package.json [hunk 2]
```diff
@@ -17,8 +17,8 @@
     "deploy:all:production": "yarn deploy:customer1:production && yarn deploy:customer2:production",
     "deploy:all:production:simple": "yarn deploy:customer1:production:simple && yarn deploy:customer2:production:simple",
     "deploy:all:all": "yarn deploy:all && yarn deploy:all:production",
-    "deploy:ci:gha": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.staging.json",
-    "deploy:ci:gha:production": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.production.json --prod",
+    "deploy:ci:gha": "yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.staging.json",
+    "deploy:ci:gha:production": "CUSTOMER_REF=${CUSTOMER_REF} yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.production.json --prod",
```
Comment (Vadorequest) [inline package.json]: Why `CUSTOMER_REF=${CUSTOMER_REF}` ? Doesn't seem necessary, and it's confusing because the above command doesn't define it.

**Review by Vadorequest — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-vercel-staging.yml (hunk 1 above)
Comment (Demmonius) [reply .github/workflows/deploy-vercel-staging.yml]: Yes I agree

File: package.json (hunk 2 above)
Comment (Demmonius) [reply package.json]: Mistake, I removed it from staging but not production

File: .github/workflows/deploy-vercel-production.yml [hunk 3]
```diff
@@ -39,6 +39,7 @@ on:
       - 'master'
       - 'main'
       - 'v2-mst-aptd-at-lcz-sty' # XXX Name of the NRN preset branch, acting as a main branch (all commits pushed to the preset branch are deployed to production)
+      - 'implement-own-vercel-deploy-action'
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-production.yml]: TODO remove

File: .github/workflows/deploy-vercel-production.yml [hunk 4]
```diff
@@ -119,57 +120,15 @@ jobs:
...
-              # $VERCEL_DEPLOYMENT_ALIASES is null, this happens when it was not defined in the vercel.json file
-              echo "There are no more aliases to configure. You can add more aliases from your vercel.json 'alias' property. See https://vercel.com/docs/configuration?query=alias%20domain#project/alias"
-              echo "$VERCEL_DEPLOYMENT_ALIASES"
-            fi
-          fi
-
+        uses: UnlyEd/github-action-deploy-on-vercel@latest
+        with:
+          command: "yarn deploy:ci:gha:production --token ${{ secrets.VERCEl_TOKEN }}"
+          apply_domain_aliases: true # Automatically create aliases from Vercel config file
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-production.yml]: renamed

File: .github/workflows/deploy-vercel-staging.yml [hunk 5]
```diff
@@ -119,92 +119,15 @@ jobs:
...
-              # $VERCEL_DEPLOYMENT_ALIASES is null, this happens when it was not defined in the vercel.json file
-              echo "There are no more aliases to configure. You can add more aliases from your vercel.json 'alias' property. See https://vercel.com/docs/configuration?query=alias%20domain#project/alias"
-              echo "$VERCEL_DEPLOYMENT_ALIASES"
-            fi
-          fi
-
+        uses: UnlyEd/github-action-deploy-on-vercel@latest
+        with:
+          command: "yarn deploy:ci:gha --token ${{ secrets.VERCEl_TOKEN }}"
+          apply_domain_aliases: true # Automatically create aliases from Vercel config file
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: renamed

File: .github/workflows/deploy-vercel-storybook.yml [hunk 6]
```diff
@@ -84,56 +84,14 @@ jobs:
...
-          else
-            # $VERCEL_DEPLOYMENT_ALIASES is null, this happens when it was not defined in the vercel.json file
-            echo "There are no more aliases to configure. You can add more aliases from your vercel.json 'alias' property. See https://vercel.com/docs/configuration?query=alias%20domain#project/alias"
-            echo "$VERCEL_DEPLOYMENT_ALIASES"
-          fi
+      - name: Deploying on Vercel
+        uses: UnlyEd/github-action-deploy-on-vercel@latest
+        with:
+          command: "yarn deploy:sb:gha --token ${{ secrets.VERCEl_TOKEN }}"
+          apply_domain_aliases: true # Automatically create aliases from Vercel config file
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-storybook.yml]: renamed

**Review by Vadorequest — CHANGES_REQUESTED** [review]

**Pull request merged**